src/hpc/thread.c

```diff
@@ -1309,73 +1309,82 @@ Region * CurrentRegion(void)
     return TLS(currentRegion);
 }
 
-#ifdef VERBOSE_GUARDS
-
-static void PrintGuardError(char *       buffer,
-                            char *       mode,
-                            Obj          obj,
-                            const char * file,
-                            unsigned     line,
-                            const char * func,
-                            const char * expr)
-{
-    sprintf(buffer, "No %s access to object %llu of type %s\n"
-                    "in %s, line %u, function %s(), accessing %s",
-            mode, (unsigned long long)(UInt)obj, TNAM_OBJ(obj), file, line,
-            func, expr);
-}
-void WriteGuardError(Obj          o,
-                     const char * file,
-                     unsigned     line,
-                     const char * func,
-                     const char * expr)
-{
-    char * buffer = alloca(strlen(file) + strlen(func) + strlen(expr) + 200);
-    ImpliedReadGuard(o);
-    if (TLS(DisableGuards))
-        return;
-    SetGVar(&LastInaccessibleGVar, o);
-    PrintGuardError(buffer, "write", o, file, line, func, expr);
-    ErrorMayQuit("%s", (UInt)buffer, 0L);
-}
+#ifdef DEBUG_GUARDS
+extern GVarDescriptor GUARD_ERROR_STACK;
+#endif
+
+// These are temporary debugging functions.
+
+static Int NumReadErrors = 0;
```

Comment:
What is "temporary" here? I guess these two variables (as they aren't ever read), and the stuff in DEBUG_GUARDS? Might be useful to be a bit clearer what we expect to keep/lose around here.

Reply:
I would actually like to leave them around and will document them. I had introduced them to deal with a large bulk of guard errors that occurred after introducing the new mechanism. `NumReadErrors` and `NumWriteErrors` allowed me to track progress on making them, the `#ifdef DEBUG_GUARDS` define allows you to inspect the C stack at the point of error from within GAP without having to fire up a debugger first.

Reply:
@ChrisJefferson Reimer improved the comments; better now?